src/components/EmployeeList/EmployeeList.tsx

```diff
@@ -96,14 +100,67 @@ const EmployeeListItem: Component<EmployeeRow> = ({ employee, contracts, actions
 
 const EmployeeList: Component<Props> = ({ state, actions }) => {
   const { filterString } = state.view.employees;
-  const employees = state.employee.list || [];
+  const employees = state.employee.list;
   const contracts = state.contract.list;
+  const allocations = state.allocation.list;
+  const projects = state.project.list;
+
   const filteredEmployees = filterEmployees(employees, filterString);
 
+  const employeesProjectMap: Map<number, Set<ProjectModel>> = new Map();
+  const employeesContractMap: Map<number, ContractModel | undefined> = new Map();
+
+  allocations.forEach((allocation) => {
+    const { projectId, contractId } = allocation;
+
+    const contract = contracts.find(c => c.id === contractId);
+    const project = projects.find(p => p.id === projectId);
+
+    if (contract && project) {
+      const employeeId = contract.employeeId;
+      if (!employeesProjectMap.has(employeeId)) {
+        employeesProjectMap.set(employeeId, new Set())
+      }
+
+      const set = employeesProjectMap.get(employeeId)!;
+      set.add(project);
+    }
+  });
+
+  projects.forEach((project) => {
+    const { projectManagerId } = project;
+
+    if (project) {
+      if (!employeesProjectMap.has(projectManagerId)) {
+        employeesProjectMap.set(projectManagerId, new Set())
+      }
+
+      const set = employeesProjectMap.get(projectManagerId)!;
+      set.add(project);
+    }
+  });
+
+  employees.forEach((employee) => {
+    const sortedContracts = contracts.filter(c => c.employeeId == employee.id).sort((a, b) => moment(a.endDate, "YYY-MM-DD").diff(moment(b.endDate, "YYY-MM-DD")))
+    var relevantContract: ContractModel | undefined = undefined;
+    if (sortedContracts.length > 0) {
+      relevantContract = sortedContracts.reduce((prev, curr) => {
+        if (moment().isAfter(curr.endDate)) {
+          return curr;
+        } else {
+          return prev ? prev : curr;
+        }
+      });
+    }
+    employeesContractMap.set(employee.id, relevantContract);
+  });
+
   const createEmployeeListItem = (employee: EmployeeModel) => {
     const employeeContracts = contracts!.filter(contract => contract.employeeId === employee.id);
+    const projects = employeesProjectMap.get(employee.id);
+    const relevantContract = employeesContractMap.get(employee.id);
 
-    return <EmployeeListItem key={employee.id} employee={employee} contracts={employeeContracts} actions={actions} />;
+    return <EmployeeListItem key={employee.id} employee={employee} projects={projects} contract={relevantContract} contracts={employeeContracts} actions={actions} />;
```

Comment:
Indentation of properties:
+ Does it need both `contracts` and `contract`?
```
<EmployeeListItem
  key={employee.id}
  employee={employee}
  projects={projects}
  contract={relevantContract}
  contracts={employeeContracts}
  actions={actions}
/>
```

Reply:
I know, it's not pretty. But computing the contractId and getting it our of contracts in the EmployeeListItem would be even more complicated.